weatherApp/Core/URLSession/URLSession+Extension.swift

```diff
@@ -0,0 +1,41 @@
+//
+//  URLSession+Extension.swift
+//  weatherApp
+//
+//  Created by Dario Nikolic on 01/09/2020.
+//  Copyright © 2020 Dario Nikolic. All rights reserved.
+//
+
+import Foundation
+
+extension URLSession {
+
+    typealias DataTaskCompletion = (Result<(HTTPURLResponse, Data), Error>) -> Void
+
+    func dataTask(with request: URLRequest, completion: @escaping DataTaskCompletion) -> URLSessionDataTask {
+        return self.dataTask(with: request) { (data, response, error) in
```

Comment:
👍  This looks great. You moved this in the core layer and you didn't use Rx here because this can be reused in another project which maybe doesn't use rxSwift.